ui/components-react/src/components-react/filter-builder/FilterBuilder.tsx

```diff
@@ -45,15 +47,62 @@ export interface PropertyFilterBuilderRuleRenderingContextProps {
   disablePropertySelection?: boolean;
 }
 
+function getGroupRuleItem(filter: PropertyFilter, parentId: string): PropertyFilterBuilderRuleGroupItem {
+  const id = Guid.createValue();
+  if (isPropertyFilterRuleGroup(filter))
+    return {
+      id,
+      groupId: parentId,
+      operator: filter.operator,
+      items: filter.rules.map((rule) => getGroupRuleItem(rule, id)),
+    };
+  else {
+    return getSingleRuleItem(filter, id);
+  }
+}
+
+function getSingleRuleItem(filter: PropertyFilterRule, parentId: string) {
+  return {
+    id: Guid.createValue(),
+    groupId: parentId,
+    property: filter.property,
+    operator: filter.operator,
+    value: filter.value,
+  };
+}
+
+function convertFilterToState(filter?: PropertyFilter): PropertyFilterBuilderState | undefined {
```

Comment:
I think conversion from filtrer to state should be moved to `FilterBuilderState` source file